6주차/mission/src/hooks/useThrottle.ts

```
@@ -0,0 +1,80 @@
+ */
+// value 타입 스로틀링
+export function useThrottle<T>(value: T, interval: number): T;
+// 콜백(함수) 타입 스로틀링
+export function useThrottle<T extends (...args: any[]) => any>(fn: T, interval: number): T;
+export function useThrottle<T>(input: T, interval: number): any {
+  // 콜백이면 함수 스로틀링, 아니면 값 스로틀링
+  if (typeof input === "function") {
+    const fn = input as (...args: any[]) => any;
+    const lastExecuted = useRef<number>(0);
+    const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+    const savedArgs = useRef<any[]>([]);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)